src/main/java/org/jenkinsci/plugins/ghprb/GhprbPullRequestMerge.java

```diff
@@ -137,44 +141,58 @@ public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, final Build
 
         // ignore comments from bot user, this fixes an issue where the bot would auto-merge
         // a PR when the 'request for testing' phrase contains the PR merge trigger phrase and
-        // the bot is a member of a whitelisted organisation
+        // the bot is a member of a whitelisted organization
         if (helper.isBotUser(triggerSender)) {
             logger.println("Comment from bot user " + triggerSender.getLogin() + " ignored.");
             return false;
         }
 
-        boolean merge = true;
+        boolean intendToMerge = false;
+        boolean canMerge = true;
         String commentBody = cause.getCommentBody();
 
-        if (isOnlyAdminsMerge() && (triggerSender == null || !helper.isAdmin(triggerSender) )) {
-            merge = false;
-            logger.println("Only admins can merge this pull request, " + triggerSender.getLogin() + " is not an admin.");
-            commentOnRequest(String.format("Code not merged because %s is not in the Admin list.", triggerSender.getName()));
-        }
-
-        if (isOnlyTriggerPhrase() && (commentBody == null || !helper.isTriggerPhrase(cause.getCommentBody()) )) {
-            merge = false;
+        // If merge can only be triggered by a comment and there is a
+        // comment
+        if (commentBody == null || !helper.isTriggerPhrase(commentBody)) {
             logger.println("The comment does not contain the required trigger phrase.");
+        }
+        else {
+            intendToMerge = true;
+        }
 
-            commentOnRequest(String.format("Please comment with '%s' to automerge this request", trigger.getTriggerPhrase()));
+        // If there is no intention to merge there is no point checking
+        if (intendToMerge && isOnlyAdminsMerge() && (triggerSender == null || !helper.isAdmin(triggerSender))) {
+            canMerge = false;
+            logger.println("Only admins can merge this pull request, "
+                    + (triggerSender != null ? triggerSender.getLogin() + " is not an admin"
+                            : " and build was triggered via automation") + ".");
+            if (triggerSender != null) {
```

Comment:
I would want to know that the build failed because the triggerSender was null.  That would tell me there was a configuration issue, or that the build was triggered by something else.